### PR TITLE
[Components with more than 1 service] LPS-192612 Refactor ModelListener components part II

### PR DESCRIPTION
**Pull request by MrJohn1911**

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:sf

File: modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/util/AnalyticsModelUtil.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+				attributeNames.add(name);
+			}
+		}
+
+		return attributeNames;
+	}
+
+	public static boolean isCustomField(
+		ExpandoTableLocalService expandoTableLocalService, long classNameId,
+		long tableId) {
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/util/AnalyticsModelUtil.java:54]: UnsafeFunction<Long, ExpandoTable> getExpandoTableFunction

File: modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/util/AnalyticsModelUtil.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+		try {
+			ExpandoTable expandoTable = expandoTableLocalService.getTable(
+				tableId);
+
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/util/AnalyticsModelUtil.java:59]: getExpandoTableFunction.apply(tableId)

File: modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ExpandoColumnModelListener.java [hunk 3]
```diff
@@ -104,13 +105,20 @@ protected String getPrimaryKeyName() {
...
+					Organization.class.getName()),
+				expandoColumn.getTableId())) {
 
 			return false;
 		}
 
-		if (isCustomField(User.class.getName(), expandoColumn.getTableId())) {
+		if (AnalyticsModelUtil.isCustomField(
+				expandoTableLocalService,
+				classNameLocalService.getClassNameId(User.class.getName()),
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ExpandoColumnModelListener.java]: expandoTableLocalService::getTable

File: modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ContactModelListener.java [hunk 4]
```diff
@@ -64,13 +45,8 @@ protected boolean isExcluded(Contact contact) {
...
 
-	private static final List<String> _attributeNames = Arrays.asList(
-		"birthday", "classNameId", "classPK", "companyId", "createDate",
-		"emailAddress", "employeeNumber", "employeeStatusId", "facebookSn",
-		"firstName", "hoursOfOperation", "jabberSn", "jobClass", "jobTitle",
-		"lastName", "male", "middleName", "modifiedDate", "parentContactId",
-		"prefixListTypeId", "skypeSn", "smsSn", "suffixListTypeId", "twitterSn",
-		"userId", "userName");
+	@Reference(target = "(entity.model.type=contact)")
+	private AnalyticsEntityModel<Contact> _contactEntityModel;
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ContactModelListener.java:27]: rename

File: modules/apps/analytics/analytics-message-sender-api/src/main/java/com/liferay/analytics/message/sender/model/listener/AnalyticsEntityModel.java [hunk 5]
```diff
@@ -25,12 +25,12 @@ public void addAnalyticsMessage(
...
 	public long[] getMembershipIds(User user) throws Exception;
 
-	public String getModelClassName();
-
-	public void syncAll(long companyId) throws Exception;
-
 	public T getModel(long id) throws Exception;
 
+	public String getModelClassName();
+
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-api/src/main/java/com/liferay/analytics/message/sender/model/listener/AnalyticsEntityModel.java:31]: just squash this with some previous commit

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Pull request closed without merging**